text/0006-folio-distributed-vs-centralized-configration.md

```diff
@@ -0,0 +1,93 @@
+- Contributors:
+  - [Julian Ladisch](julian.ladisch@gbv.de)
+  - [Florian Gleixner](Florian.Gleixner@lrz.de)
+- RFC PRs: 
+  - PRELIMINARY REVIEW: https://github.com/folio-org/rfcs/pull/23
+  - DRAFT REVIEW: TBD
+  - PUBLIC REVIEW: TBD
+  - FINAL REVIEW: TBD
+- Outcome: (Leave this blank.  Will eventually be either ACCEPTED or REJECTED)
+
+
+# Folio distributed vs. centralized configuration
+
+## Summary
+
+Drop mod-configuration module. Favour distributed configuration over centralized configuration, provide guidelines for storage of configuration values.
+
+## Motivation
+
+FOLIO relies on a tenant aware extendable microservice system. It is highly configurable and adoptable on all kind of requirements. FOLIO had no strict guidelines for developers, how and where to store the configuration values. This led to a situation where different approaches where developed by different developer teams.
+This makes it difficult for implementers, administrators and users to configure the system appropriate, as they may have to search the documentation how to read or write these configuration values.
+Basically there are two competing concepts where to store the values: distributed in the owning modules or centralized in a special module that offers a configuration store for other modules. There exist two solutions for centralized configurations: mod-settings and mod-configuration (deprecated, to be dropped).
+This RFC shall give guidelines for developers, where configuration values shall be set.
+
+## Detailed Explanation/Design
+
+This RFC deals with configurations that configure the behaviour of the Folio tenant. Among all different kinds of configurations, this RFC does not deal with the following configurations, as their location and method to set/get them shall not be changed:
+
+* Settings stored in Okapi's /_/env APIs.
+* Settings stored in Infrastructure (Kubernetes / Rancher) config maps and secrets.
+* Settings stored in module container environment variables.
+* Settings stored in the stripes front-end (stripes.config.js, etc.)
+
+### mod-configuration will be dropped until Ramsons release
+
+mod-configuration is deprecated due to security problems since March 2022. It shall not be used any more to add new configuration variables. Modules still using mod-configuration have to move to other solutions until the Ramsons release.
```

Comment:
> Modules still using mod-configuration have to move to other solutions until the Ramsons release

Maybe `Modules still using mod-configuration have until the Ramsons release to move to other solutions`

Reply:
I am not a native english speaker, but is there any difference?

Reply:
@okolawole-ebsco is it OK, to resolve this conversation?